### ISSUE TITLE
Cannot modify header information - headers already sent

### ISSUE THREAD

**Issue by tacruc** [nextcloud/server#7423]
Not shure if the error is Produced by the server or by passman.

### Steps to reproduce
1. visit your nextcloud

### Expected behaviour
Not placing this error in the log.

### Actual behaviour
Producing the error

> Error PHP Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69 a few seconds ago

## Server configuration detail

**Operating system:** Linux 4.9.0-3-amd64 #1 SMP Debian 4.9.30-2+deb9u3 (2017-08-06) x86_64

**Webserver:** Apache/2.4.27 (Debian) (apache2handler)

**Database:** mysql 10.1.26

**PHP version:** 7.0.19-1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, apache2handler, mysqlnd, PDO, xml, apcu, apc, bz2, calendar, ctype, curl, dom, enchant, mbstring, fileinfo, ftp, gd, gettext, gmp, gnupg, iconv, igbinary, imagick, imap, interbase, intl, json, ldap, exif, mcrypt, mongodb, mysqli, odbc, pdo_dblib, PDO_Firebird, pdo_mysql, PDO_ODBC, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, pspell, readline, recode, redis, shmop, SimpleXML, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, wddx, xmlreader, xmlrpc, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 13.0.0 Beta 1 - 13.0.0.6

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - activity: 2.6.1
 - admin_audit: 1.3.0
 - admin_notifications: 1.0.1
 - announcementcenter: 3.1.1
 - bookmarks: 0.10.1
 - calendar: 1.5.6
 - caniupdate: 0.1.1
 - circles: 0.13.6
 - cms_pico: 0.9.6
 - comments: 1.3.0
 - contacts: 2.0.1
 - dav: 1.4.5
 - end_to_end_encryption: 1.0.1
 - external: 3.0.0
 - federatedfilesharing: 1.3.1
 - federation: 1.3.0
 - files: 1.8.0
 - files_accesscontrol: 1.3.0
 - files_automatedtagging: 1.3.0
 - files_downloadactivity: 1.2.0
 - files_external: 1.4.1
 - files_markdown: 2.0.1
 - files_pdfviewer: 1.2.0
 - files_sharing: 1.5.0
 - files_texteditor: 2.5.1
 - files_trashbin: 1.3.0
 - files_versions: 1.6.0
 - files_videoplayer: 1.2.0
 - firstrunwizard: 2.2.1
 - gallery: 18.0.0
 - groupfolders: 1.1.0
 - issuetemplate: 0.3.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.1.0
 - mail: 0.7.5
 - metadata: 0.6.0
 - mood: 0.3.3
 - news: 11.0.5
 - nextcloud_announcements: 1.2.0
 - notes: 2.3.2
 - notifications: 2.1.0
 - oauth2: 1.1.0
 - ocdownloader: 1.5.4
 - passman: 2.1.4
 - password_policy: 1.3.0
 - piwik: 0.3.2
 - polls: 0.8.0
 - provisioning_api: 1.3.0
 - qownnotesapi: 17.5.0
 - quota_warning: 1.2.0
 - ransomware_protection: 1.1.0
 - serverinfo: 1.3.0
 - sharebymail: 1.3.0
 - spreedme: 0.3.11
 - survey_client: 1.1.0
 - systemtags: 1.3.0
 - telephoneprovider: 1.0.0
 - theming: 1.4.1
 - twofactor_backupcodes: 1.2.3
 - twofactor_u2f: 1.4.0
 - updatenotification: 1.3.0
 - workflowengine: 1.3.0
Disabled:
 - bruteforcesettings
 - checksum
 - deck
 - drawio
 - encryption
 - files_clipboard
 - files_frommail
 - files_opds
 - files_reader
 - files_retention
 - gpxedit
 - gpxmotion
 - gpxpod
 - impersonate
 - keeweb
 - nextant
 - ocsms
 - ojsxc
 - onlyoffice
 - phonetrack
 - previewgenerator
 - radio
 - socialsharing_diaspora
 - socialsharing_email
 - socialsharing_facebook
 - socialsharing_googleplus
 - socialsharing_twitter
 - spreed
 - tasks
 - twofactor_totp
 - user_external
 - user_ldap
 - weather

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITVE VALUE***",
        "***REMOVED SENSITVE VALUE***"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "tempdirectory": "\/tmp\/",
    "filelocking.enabled": "true",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0,
        "timeout": 0
    },
    "dbtype": "mysql",
    "version": "13.0.0.6",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "forcessl": true,
    "forceSSLforSubdomains": true,
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauth": 1,
    "mail_smtpport": "587",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpsecure": "tls",
    "mail_smtpauthtype": "PLAIN",
    "loglevel": 2,
    "maintenance": false,
    "enable_previews": true,
    "enabledPreviewProviders": [
        "OC\\Preview\\PNG",
        "OC\\Preview\\JPEG",
        "OC\\Preview\\GIF",
        "OC\\Preview\\BMP",
        "OC\\Preview\\XBitmap",
        "OC\\Preview\\MP3",
        "OC\\Preview\\TXT",
        "OC\\Preview\\MarkDown",
        "OC\\Preview\\TIFF",
        "OC\\Preview\\SVG",
        "OC\\Preview\\RAW",
        "OC\\Preview\\Epub",
        "OC\\Preview\\FB2",
        "OC\\Preview\\PDF",
        "OC\\Preview\\OpenDocument",
        "OC\\Preview\\StarOffice",
        "OC\\Preview\\MSOfficeDoc",
        "OC\\Preview\\MSOffice2003",
        "OC\\Preview\\MSOffice2007"
    ],
    "preview_max_x": 3840,
    "preview_max_y": 2160,
    "preview_max_scale_factor": 1,
    "appstore.experimental.enabled": true,
    "remember_login_cookie_lifetime": 1296000,
    "session_lifetime": 86400,
    "session_keepalive": true,
    "trashbin_retention_obligation": "auto",
    "knowledgebaseenabled": true,
    "logtimezone": "Europe\/Berlin",
    "cron_log": true,
    "log_rotate_size": 10857600,
    "theme": "",
    "updatechecker": false,
    "htaccess.RewriteBase": "\/",
    "check_for_working_wellknown_setup": true,
    "check_for_working_htaccess": true,
    "appstoreenabled": true,
    "updater.release.channel": "beta",
    "singleuser": false,
    "overwrite.cli.url": "HIDEN"
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

## Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:57.0) Gecko/20100101 Firefox/57.0

**Operating system:** 

## Logs

<details><summary>Nextcloud log</summary>

Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	a few seconds ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	a few seconds ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	a few seconds ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	a few seconds ago
Error	no app in context	issuetemplate	a few seconds ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	6 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	6 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	6 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	6 minutes ago
Error	PHP	Array to string conversion at /var/www/nextcloud/apps/issuetemplate/lib/DetailManager.php#104	6 minutes ago
Error	no app in context	issuetemplate	6 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	no app in context	issuetemplate	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	15 minutes ago
Error	no app in context	issuetemplate	15 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	no app in context	issuetemplate	19 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/nextcloud/apps/passman/personal.php:1) at /var/www/nextcloud/lib/private/AppFramework/Http/Output.php#69	19 minutes ago
Error	no app in context	issuetemplate	19 minutes ago
</details>

**Comment by tacruc** [comment]
Is actually a bug in Passman.
Created a PR to fix it
https://github.com/nextcloud/passman/pull/415

**Comment by borekon** [comment]
I'm having the same issue.
Disabling passman removes the error. 
Maybe it's not yet fixed

**Comment by tacruc** [comment]
But it is a bug in Passman. I open a Issue and PR which fixed it, but its not merged till now

**Comment by wjwieland** [comment]
I am getting deluged with this log entry as well.  Is it posible that this is a problem with the web-extension instead of the server side module?  There is some weirdness with the extension right now that I opened an issue for on the web-extension, but got the 'its must be a browser issue' response.  Based on what I am seeing, I think it is more than that....

Anyway, it would be good to see some action on this.

**Comment by ccall48** [comment]
Has this fix been merged yet? I've been getting this error for a year or so thought i'd check it out because its starting to get annoying this error filling up my log - I am on the latest stable version of nextcloud (13 I think).

**Comment by tacruc** [comment]
No the https://github.com/nextcloud/passman/pull/415 got closed.

**Comment by danielkesselberg** [comment]
https://github.com/nextcloud/passman/blob/master/personal.php is fixed but there is no release yet. Please keep this discussion in https://github.com/nextcloud/passman :+1:

**Comment by wjwieland** [comment]
So, the bottom line is that the file located at:
../nextcloud/apps/passman/personal.php
needs to be changed.

Mine looked as follows:


Which I changed to:
<?php

Nothing more.  I fail to see why this could not have been stated clearly somewhere by the maintainers, who apparently fixed the issue, but did not merge nor propagate it.  At any rate, I made this change and my log is no longer filling up with this idiotic message.  Hope this helps someone out there...
